### PR TITLE
heroic: 2.9.2 -> 2.10.0

### LINKED ISSUE #262299: Heroic Launcher gog's games won't install
### Describe the bug
Heroic Launcher while trying to install a game from gog will download it, but won't install it.

You can see downloaded installer inside your `${HOME}/.cache/heroicGOGdl` directory.

After "installation", launcher will say on the game's details screen that `Game not available`

### Expected behavior
It should install.

### Additional context
You can't just use the gog installer like in a normal linux distribution:
```console
[user@system:~]$ sh .cache/heroicGOGdl/dead_cells_1_24_4_65630.s
Verifying archive integrity... All good.
Uncompressing Dead Cells (GOG.com)  100%
Collecting info for this system...
Operating system: linux
CPU Arch: x86_64
trying mojosetup in bin/linux/x86_64
USING en_US
./startmojo.sh: 55: /tmp/selfgz4843828657/bin/linux/x86_64/mojosetup: not found
Error: Couldn't run mojosetup
.cache/heroicGOGdl/dead_cells_1_24_4_65630.s: line 517: /bin/rm: No such file or directory
```

So you need to use **steam-run** to install it.

### Notify maintainers
@aidalgol

### Metadata
```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.58, NixOS, 23.05 (Stoat), 23.05.20231018.80c1aab`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.5`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**heroic-launcher**: 2.9.2 - unstable nixpkgs

### LINKED ISSUE #264156: Build failure: heroic
### Steps To Reproduce

1. add `heroic` to `environment.systemPackages` on unstable
2. `sudo nixos-rebuild switch --upgrade`

### Build log

```
   … while evaluating derivation 'heroic-unwrapped-2.9.2'
         whose name attribute is located at /nix/store/bmcm18ngcq3pplprdkypa348qmi8lcps-nixos/nixos/pkgs/stdenv/generic/make-derivation.nix:300:7

       … while evaluating attribute 'installPhase' of derivation 'heroic-unwrapped-2.9.2'

         at /nix/store/bmcm18ngcq3pplprdkypa348qmi8lcps-nixos/nixos/pkgs/games/heroic/default.nix:75:3:

           74|   # https://github.com/electron/electron/issues/32317
           75|   installPhase = let
             |   ^
           76|     binPlatform = if stdenv.isDarwin then "darwin" else "linux";

       … from call site

         at /nix/store/bmcm18ngcq3pplprdkypa348qmi8lcps-nixos/nixos/pkgs/stdenv/generic/check-meta.nix:449:16:

          448|         {
          449|           no = handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; };
             |                ^
          450|           warn = handleEvalWarning { inherit meta attrs; } { inherit (validity) reason errormsg; };

       … while calling 'handleEvalIssue'

         at /nix/store/bmcm18ngcq3pplprdkypa348qmi8lcps-nixos/nixos/pkgs/stdenv/generic/check-meta.nix:226:38:

          225|
          226|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          227|     let

       error: Package ‘electron-24.8.6’ in /nix/store/bmcm18ngcq3pplprdkypa348qmi8lcps-nixos/nixos/pkgs/development/tools/electron/binary/generic.nix:35 is marked as insecure, refusing to evaluate.

```

### Additional context

The package is locked to Electron 24 by this PR it seems https://github.com/NixOS/nixpkgs/pull/254075 but Electron 24 is EOL now so it doesn't build

### Notify maintainers

@aidalgol 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[fazo@namyra:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.5.7, NixOS, 23.11 (Tapir), 23.11pre537656.7c9cc5a6e5d3`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.17.1`
 - channels(root): `"home-manager, nixos"`
 - channels(fazo): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by aidalgol**
## Description of changes
- Update Heroic Games Launcher to latest stable release (fixes #264156).
- Update Nile Amazon Games client, used by Heroic.
- Heroic FHS env fix (closes #262299).

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
I can't log in to Epic with this version - it just doesn't seem to recognize the account exists, even though Legendary seems to be aware of it (via the CLI)

**Comment by aidalgol** [comment]
I got an odd error about my Epic session when I first launched 2.10, and I had to log out of and log back in to my Epic account, dismissing error popups along the way.  This is not the first time I've had session breakages when upgrading.

**Comment by K900** [comment]
I even cleared out ~/.config/heroic and it's still busted. I'll see if there's anything interesting in the logs later.

File: pkgs/games/heroic/fhsenv.nix [hunk 1]
```diff
@@ -29,7 +29,10 @@ buildFHSEnv {
     perl
     psmisc
     python3
+    unzip
     which
+    wine
```
Comment (SuperSamus) [inline pkgs/games/heroic/fhsenv.nix:34]: I don't agree with adding Wine in the FHS env:
- That Wine specifically is stable Wine, which isn't really recommended for games (for instance, all Unity games suffer from this bug: https://bugs.winehq.org/show_bug.cgi?id=48121)
- If someone already has a version of Wine installed system-wide, they'll have to forcefully install another version that they may not want to use
- Heroic allows you to install whatever Wine version you want, what's the point in including one here?

Also, it was removed from Lutris #224545 and Bottles #202265.

**Comment by yeoldegrove** [comment]
> I can't log in to Epic with this version - it just doesn't seem to recognize the account exists, even though Legendary seems to be aware of it (via the CLI)

I also had issues when trying to login. But after quitting heroic and starting it again, I am successfully logged in.

**Comment by equirosa** [comment]
Result of `nixpkgs-review pr 265555` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>heroic</li>
    <li>heroic-unwrapped</li>
    <li>legendary-gl</li>
    <li>legendary-gl.dist</li>
    <li>nile</li>
    <li>nile.dist</li>
    <li>rare</li>
    <li>rare.dist</li>
  </ul>
</details>

**Pull request merged**